DataFormats/L1TrackTrigger/interface/TTDTC.h

```diff
@@ -0,0 +1,74 @@
+#ifndef __L1TrackTrigger_TTDTC__
+#define __L1TrackTrigger_TTDTC__
+
+#include "DataFormats/L1TrackTrigger/interface/TTTypes.h"
+#include "DataFormats/L1TrackTrigger/interface/TTBV.h"
+
+#include <bitset>
+#include <utility>
+#include <vector>
+
+/*! 
+ * \class  TTDTC
+ * \brief  Class to store hardware like structured TTStub Collection used by Track Trigger emulators
+ * \author Thomas Schuh
+ * \date   2020, Jan
+ */
+class TTDTC {
+public:
+  // outer tracker DTC types
+  typedef std::bitset<TTBV::S> BV;         // bit accurate Stub
+  typedef std::pair<TTStubRef, BV> Frame;  // TTStub with bit accurate Stub
+  typedef std::vector<Frame> Stream;       // stub collection transported over an optical link between DTC and TFP
+  typedef std::vector<Stream> Streams;     // collection of optical links
+
+public:
+  TTDTC() {}
+
+  TTDTC(const int& numRegions, const int& numOverlappingRegions, const int& numDTCsPerRegion);
+
+  ~TTDTC() {}
+
+  // Access to product configurations
+  int numRegions() const;
+  int numDTCBoards() const;
+  int numDTCChannel() const;
+  int numTFPChannel() const;
+
+  // Access to one specific stream of TTStubRefs using DTC identifier (region[0-8], board[0-23], channel[0-1])
+  // dtcRegions aka detector regions are defined by tk layout
+  Stream& dtcStream(const int& dtcRegion, const int& dtcBoard, const int& dtcChannel);
```

Comment:
This function should be `const` and return `const Stream&`.

Reply:
I was to lazy to write a set and get function, therefore the non const reference. Shall I write set and get separated instead?

Reply:
Yes, separate functions are preferred (note 2.10 and 2.11 in http://cms-sw.github.io/cms_coding_rules.html). Note that currently this function can not be used to get (because it is not `const`).